src/plots/cartesian/axes.js

```diff
@@ -527,7 +536,7 @@ axes.prepTicks = function(ax) {
         if(ax.tickmode === 'array') nt *= 100;
 
 
-        ax._roughDTick = (Math.abs(rng[1] - rng[0]) - (ax._lBreaks || 0)) / nt;
```

Comment:
Looking much nicer now!

I'm just noticing that the auto `dtick` we get depends on whether there's a break within the visible range or not, which seems weird, and seems like exactly what this `- ax._lBreaks` was designed to avoid. For example (back on everyone's favorite, `axes_breaks-candlestick2` 😅) if I zoom in while keeping a break in range I can't get `dtick` less than 2 hours:
<img width="911" alt="Screen Shot 2020-06-02 at 10 55 55 PM" src="https://user-images.githubusercontent.com/2678795/83590721-629bba00-a524-11ea-8a02-1c9cb7d99063.png">

But shift the break off the edge with exactly the same scale and it snaps to 15 min:
<img width="897" alt="Screen Shot 2020-06-02 at 10 57 52 PM" src="https://user-images.githubusercontent.com/2678795/83590806-937bef00-a524-11ea-8c22-a01c30a2d7e0.png">

Is there something bad that happens if this line is left as it was?

Reply:
Good catch. Addressed in https://github.com/plotly/plotly.js/pull/4831/commits/9241578b038d0a8d9adcdd3eaa5b49baf496306c.